Refactoring the code to keep the database open for entire time, getSession(), reload()

Hopeberg opened the pull request:
#46 
connect_db() is called once in main 
getSession() returns the current session
getUser() returns the current user and user_id 
reload() reloads a page upon a form input error in register/login
removed the use of before_request and after_request 
* When you sign in, the template does not switch the top links. Always shows sign in when it is supposed to show username. FIXED 

* When you go to another profile the template says logout with not your name. It shows name of the user you are watching. FIXED 

* When visiting user you can post a message looking like you are posting from their profile. FIXED

* Console is always printing error "Error when trying to get the profile user in database ..." line 361 FIXED

danielgron reviewed (commented):
There's too much for me to go through in a sitting now, but I can confirm that tests are passing.

Romes8 commented:
I found some things that do not work correctly:  (Logged in as Tester for reference)
- When you sing in the template does not switch the top links. Always shows sign in when it is supposed to show username. 
- When you go to another profile the template says logout with not your name. It shows name of the user you are watching
- When visiting user you can post a message looking like you are posting from their profile. 
- Console is always printing error "Error when trying to get the profile user in database ..." line 361
- Closing DB is not preset anywhere if I am looking at it right. Maybe we can have a function for it and call if all the functions that have some queries ?

Hopeberg commented:
* When you sign in, the template does not switch the top links. Always shows sign in when it is supposed to show username. FIXED 

* When you go to another profile the template says logout with not your name. It shows name of the user you are watching. FIXED 

* When visiting user you can post a message looking like you are posting from their profile. FIXED

* Console is always printing error "Error when trying to get the profile user in database ..." line 361 FIXED

Regarding closing the db, I think it is good that we push this version. I don't think we need to open/close database as it creates overhead, but if we are forced to do it, I will make the changes in a new PR.
